### ISSUE TITLE
"Linux Air Combat for Raspberry Pi-5", an Install based App

### ISSUE THREAD

**Issue by rjbosen** [Botspot/pi-apps#2565]
### What is the name of the app?

Linux Air Combat

### Where is the app hosted?

https://sourceforge.net/projects/linuxaircombat/

### About the app

Linux Air Combat ("LAC")  is a free, open-source, online, multi-player combat flight simulator that strives for realism, with 54 of the most important aircraft of World War II. Control flight with joystick, console game controller, or mouse and keyboard.

Fighters can combat other fighters, escort friendly bombers, attack enemy bombers, or strafe enemy airfields, ships, and buildings.

Bombers can rain destruction upon enemy airfields or dive-bomb ships and facilities, but they must first survive the trip to and through enemy territory!

Bombers are automatically defended by their own gunners.

After installation, launch LAC from a bash shell by typing:

lac 

followed by the ENTER key.

### Upload file or Add PR Link

[Linux Air Combat for Rpi5.zip](https://github.com/Botspot/pi-apps/files/14671353/Linux.Air.Combat.for.Rpi5.zip)

### Confirmations

- [X] I have confirmed that this app is legal and not piracy.
- [X] I have confirmed that this app has never been discussed in https://github.com/Botspot/pi-apps/issues and https://github.com/Botspot/pi-apps/pulls, and it is not in the Pi-Apps app list.
- [X] I have read and followed the [creating an app for Pi-Apps documentation](https://pi-apps.io/wiki/development/Creating-an-app/).
- [X] I have confirmed that this app can run on all [Pi-Apps supported systems](https://pi-apps.io/install/).

**Comment by Botspot** [comment]
your zipfile is empty lol

**Comment by rjbosen** [comment]
Whoops! Let me look into that. On my end it says it's only 202 bytes. Is that what you got?

**Comment by Botspot** [comment]
yes

**Comment by Botspot** [comment]
> Sorry. This is the first time I have used the RPi5 archiver to create a zip file. Could I upload a tar instead?

sure

**Comment by rjbosen** [comment]
Thanks. Give me 5 mins or so....

**Comment by rjbosen** [comment]
OK. I have the tar file and I confirmed it untars correctly.

But now I cannot figure out how to upload it to your site. The first time I did it it was hard to figure out the process. Now it seems stuck on the prior upload. Is there another way?

**Comment by rjbosen** [comment]
Looks like it won't let me send a tar file.

I'll try again to figger out how to make a zipive....

**Comment by rjbosen** [comment]
[LinuxAirCombatForRpi5.zip](https://github.com/Botspot/pi-apps/files/14671607/LinuxAirCombatForRpi5.zip)

**Comment by rjbosen** [comment]
I just uploaded "LinuxAirCombatForRpi5.zip". Did you get 5.9 KB?

**Comment by rjbosen** [comment]
Discord won't me sign up because I do not own a cell phone. I do not WANT a cell phone. Who wants to be tracked all the time and targeted for spam, ads, and monitored by the NSA?

**Comment by Botspot** [comment]
the zip works, you can see it if you click the "`Click to show contents preview`" button here.

**Comment by rjbosen** [comment]
Yes. I see that the content seems to be exactly what I intended. I have exercised the install and uninstall scripts from my end several times and I confirm they work for me on my Raspberry Pi 5 using 64-bit Raspberry Pi OS.

**Comment by Botspot** [comment]
OK, not sure how soon I can review it. Thanks for the work.

**Comment by rjbosen** [comment]
Thank you. I'm excited about the high-quality exposure my flight simulator will have through your efforts.

Regards....

**Comment by theofficialgman** [comment]
```
# Download the Install tar file
cd
cd Downloads || error "Failed to enter the Downloads directory."
```
use the `/tmp` folder for all temporary file storage and manipulation

if this flight sim is your project as you say it is you should really improve your makefile
```
OBJS=3ds.o aiobject.o audio.o cockpit.o common.o conf.o dirs.o effects.o fileloader.o FlightModel.o gl.o glland.o land.o loader_tga.o main.o mathtab.o menu.o mission.o model.o NetworkApi.o object.o objload.o pilots.o vertexarray.o

CXX=g++ -Ofast -g0 -fPIE

LIBS= -lglut -lGLU -lSDL -lSDL_mixer -L/lib64 -lGL

Lac09p55: $(OBJS)
	mkdir -p ~/.LAC
	cp DefaultHeightMap.LAC ~/.LAC/DefaultHeightMap.LAC 
	g++ -o Lac09p55 $(LIBS) $(OBJS)
	cp Lac09p55 bin/Release/Lac09p55
	chmod +x bin/Release/Lac09p55
clean:
	@rm -f *.o
	@rm -f Lac09p55
```

most makefile on other projects have separate rules for installing than building (eg: an `install:` target). all that copying into the `/usr/` directory should be handled in the Makefile rather than having pi-apps carry that.
building also should not touch the users home directory which yours does

**Comment by theofficialgman** [comment]
makefile is incorrect. LIBS need to come after OBJS otherwise you get undefined references

**Comment by theofficialgman** [comment]
makefile produces a binary that cannot be run from the build folder. its common practice that installation is not required to run a binary and the build directory is setup such that it can be run

**Comment by theofficialgman** [comment]
`Info: callbackSwitchMultimedia() is attempting to invoke firefox.`
a function `callbackSwitchMultimedia` attempts to start firefox which is not guaranteed to be installed or the users default browser. using `x-www-browser` will open the users default browser

**Comment by rjbosen** [comment]
Thanks for helping me improve.

Perhaps the presence of source code and makefile in my download archive confused you? 

Although my standard download includes source code and an associated makefile, there is no need for their use because the bin/Release subfolder includes a precompiled executable for the Raspberry Pi 5 on 64-bit Raspberry Pi OS. The installation script I sent you actually deletes the source code and makefile and uses only the precompiled binary executable. I don't want to change my standard download archive structure for Pi-Apps because it is widely used and comprehensively documented for so many platforms.

It's true that LAC demands installation at /usr/bin and will not run correctly from the build folder. This is indeed unusual, but it is important and very helpful in maintaining compatibility with AppImage distribution. I really don't want to reverse that decision for the Pi-Apps because it would demand treating Pi-Apps differently from everybody else.

It's true that the function callbackSwitchMultimedia() is attempting to invoke firefox. LAC has a run-time configuration mechanism through which LAC users designate their own default browser for use with LAC, which may be different from the default browser they use for other activities.

**Comment by theofficialgman** [comment]
> Perhaps the presence of source code and makefile in my download archive confused you?
> Although my standard download includes source code and an associated makefile, there is no need for their use because the bin/Release subfolder includes a precompiled executable

Nope. It was clear to me that the source code and binaries both exist in the download. I am giving you feedback on the compilation process because I would rather not use your binaries when compilation is "simple" and won't result in binaries that are incompatible with the various distros that we support. If you compile on bookworm, GLIBC requirements will be too high in the binary to run it on bullseye. For applications that can be compiled quickly and easily (like yours) it is always desirable to perform the compilation on the host.

> there is no need for their use because the bin/Release subfolder includes a precompiled executable for the Raspberry Pi 5 on 64-bit Raspberry Pi OS. The installation script I sent you actually deletes the source code and makefile and uses only the precompiled binary executable.

See above

> I don't want to change my standard download archive structure for Pi-Apps because it is widely used and comprehensively documented for so many platforms.

I have no asked you to change your download archive structure.

> It's true that LAC demands installation at /usr/bin and will not run correctly from the build folder. This is indeed unusual, but it is important and very helpful in maintaining compatibility with AppImage distribution.

Appimages do not require your "lazy" build process.

> I really don't want to reverse that decision for the Pi-Apps because it would demand treating Pi-Apps differently from everybody else.

I am not asking that you fix something for pi-apps. I am asking that you fix your terrible build and install system for everyone. Please follow the GNU compliant makefile standard already in place. Having users execute a series of janky install commands is not. We do not want to maintain your janky install commands in pi-apps.

I suggest you read some documentation https://access.redhat.com/documentation/fr-fr/red_hat_enterprise_linux/9/html/packaging_and_distributing_software/placing-arbitrary-artifacts-in-the-system-using-the-make-install-command_preparing-software-for-rpm-packaging https://www.gnu.org/software/make/manual/make.html and look at other FOSS projects and investigate using automake https://www.gnu.org/software/automake/ for creating your GNU standards compliant makefile.

> It's true that the function callbackSwitchMultimedia() is attempting to invoke firefox. LAC has a run-time configuration mechanism through which LAC users designate their own default browser for use with LAC, which may be different from the default browser they use for other activities.

don't make firefox the default then. have it be `x-www-browser` or just use xdg-open if you are trying to open a webpage.

**Comment by theofficialgman** [comment]
Here is an example of a (potentially not fully standards compliant) professional makefile https://github.com/aristocratos/btop/blob/main/Makefile from btop

**Comment by rjbosen** [comment]
Is "x-www-browser" something that will work on different distros? Or is it specific to Raspberry Pi OS? I just tried it on PcLinuxOs and got "command not found".  But on Raspberry Pi Os it does work.

**Comment by theofficialgman** [comment]
> Is "x-www-browser" something that will work on different distros? Or is it specific to Raspberry Pi OS? I just tried it on PcLinuxOs and got "command not found". But on Raspberry Pi Os it does work.

its debian (and derivative) specific. hence why you should use `xdg-open put_webpage_url_here` which is universal. (I assume you are trying to open a URL, I have not read the source code for your application)

**Comment by rjbosen** [comment]
Yes. Thanks. That xdg-open open looks a lot better. I will make that change for the default in my next update.

**Comment by rjbosen** [comment]
I will update my install scripts to use /tmp instead of:

```
# Download the Install tar file
cd
cd Downloads || error "Failed to enter the Downloads directory."
```
I can see the wisdom of that. Thanks.

**Comment by rjbosen** [comment]
I think the only place where my Makefile touches the user/s home filesystem is this:

	mkdir -p ~/.LAC
	cp DefaultHeightMap.LAC ~/.LAC/DefaultHeightMap.LAC 

I remember that solved a thorny install problem for me YEARS ago. For the past 5 years or so those lines have just been leftovers. I can delete those lines right now because they haven't been needed for a long time. Thanks for helping me clean those out.

**Comment by Botspot** [comment]
Any updates on this?

**Comment by rjbosen** [comment]
Sorry... I"m not forgetting or ingnoring this..... I just had a big "life intervention" event. Please don't let my delay interfere with your other work. I'll get back to this.... I'm just not sure when.



 ---- On Mon, 01 Apr 2024 17:20:27 -0500  Botspot  wrote ---

**Comment by Botspot** [comment]
I saw [your new post](https://forums.raspberrypi.com/viewtopic.php?p=2215471&sid=3a979ddcc6f4f73f4bd13df7e6a92e85#p2215471) on the Pi forums. Any updates for here, or have you lost interest in adding this to Pi-Apps?

**Comment by rjbosen** [comment]
Thanks for the followup. I am still interested, but my progress has been slowed by an important (medical) problem. I don't know how long it will be before I can catch up. Higher priorities right now. I hope you are doing well and I remain an admirer of your product.

-Robert Bosen-
(Linux Air Combat)


 ---- On Fri, 26 Apr 2024 01:51:13 -0500  Botspot  wrote ---
